### DISCUSSION TITLE
setting title of shell window

### DISCUSSION THREAD

**Discussion by bw1faeh0** [chrisant996/clink#319] [Ideas]
Hi,

using the cmd built in command "TITLE" lets me set the title of the current cmd window. 

What would be the best approach to set the title every time I change the directory like `TITLE %cd%`
(using cd command, using .., using cd .., using every available method)?

Writing a custom lua script?
Extending a module of flexprompt?
Aliases/doskey macros?

Any ideas or suggestions?

**Comment by chrisant996** [comment]
Do you know there's an ANSI escape code to set the title?  Some people use a prompt filter to add the escape code into the prompt string.

**Reply by bw1faeh0** [reply]
Some [Microsoft Documentation](https://docs.microsoft.com/en-us/windows/console/console-virtual-terminal-sequences#window-title) mentions an escape sequence, but to be honest I don't know how to handle this.

Maybe I could create a custom flexprompt module, that does not change the prompt at all, but fires as escape sequence, that set the title?
Then I would have to enable the custom module in my flexprompt settings, to be executed when the prompt is updated?

**Reply by chrisant996** [reply]
There are many options.
Here are a few:

- Make a flexprompt module. I wouldn't recommend it, though. There are easier and more flexible options.

- Make a separate prompt filter, with a high priority such as 200. Prepend the escape code to the prompt string passed into the prompt filter. By running after flexprompt, it can add to whatever flexprompt (and other prompt filters) have constructed. And by prepending, it ensures not to interfere with the rest of the prompt.

- Use an onbeginedit event to print the escape code.

Since the onbeginedit approach is simple and small, I'll show that one here. Or I can also show the separate prompt filter approach, if you would like.

**title_cwd.lua**
```lua
clink.onbeginedit(function()
    clink.print("\x1b]2;" .. os.getcwd() .. "\x07", NONL)
end)
```

_**Caveat:** I haven't tested that. But it should work._

**Reply by bw1faeh0** [reply]
your sample is working out of the box!
Thanks a lot!
